### PR TITLE
Pmd7 upgrade 7.2.0 spring set 01

### PR DESCRIPTION
**Pull request by stokpop**
spring set 01 remarks:

`AvoidModelMapAsRenderParameter` - unit tests not complete (e.g. non-matches missing, second part of xpath expression is not covered) and what is comment for two bad ones: "[--> NOT, see JPCC-88?]"

`AvoidSpringApplicationContextRecreationTest` - assume creation of all ApplicationContexts is bad in methods - does not work in designer without proper classpath imports

PMD7 question: Why `MethodDeclaration`,`LocalVariableDeclaration`,`FieldDeclaration` but `VariableDeclarator`?

### CODE REVIEW

File: src/test/resources/com/jpinpoint/perf/lang/java/ruleset/spring/xml/AvoidModelMapAsRenderParameter.xml [hunk 1]
```diff
@@ -25,8 +26,7 @@ public class AvoidModelMapAsRenderParameterTest {
 
 	// ModelMap in session between action and render, and after that
 	// render in method name, no RenderRequest param
-	// must flag [--> NOT, see  JPCC-88?]
-	@ResourceMapping("subview-init")
+	@ResourceMapping("subview-init") // bad [--> NOT, see JPCC-88?]
```
Comment (jborgers) [inline src/test/resources/com/jpinpoint/perf/lang/java/ruleset/spring/xml/AvoidModelMapAsRenderParameter.xml:29]: Seems to me that this is found out not to be bad, see Jira issue JPCC-88. I think jira is shut down, I didn't find it.

File: src/test/resources/com/jpinpoint/perf/lang/java/ruleset/spring/xml/AvoidModelMapAsRenderParameter.xml [hunk 2]
```diff
@@ -36,29 +36,24 @@ public class AvoidModelMapAsRenderParameterTest {
...
 	private String initialOverviewRender01(final PortletRequest request,
 			final ModelMap modelMap, @RequestParam("viewName") final String viewName) {
 		modelMap.put("texts", messageSource);
 		return "/" + viewName;
 	}
 
 	// ModelMap in session between action and render, and after that
-	// must flag [--> NOT, see JPCC-88?]
-	@ResourceMapping("subview-init")
+	@ResourceMapping("subview-init") // bad [--> NOT, see JPCC-88?]
```
Comment (jborgers) [inline src/test/resources/com/jpinpoint/perf/lang/java/ruleset/spring/xml/AvoidModelMapAsRenderParameter.xml:47]: same here

File: src/test/resources/com/jpinpoint/perf/lang/java/ruleset/spring/xml/AvoidModelMapAsRenderParameter.xml (hunk 1 above)
Comment (jborgers) [reply src/test/resources/com/jpinpoint/perf/lang/java/ruleset/spring/xml/AvoidModelMapAsRenderParameter.xml:29]: and I cannot remember

**Review by jborgers — APPROVED** [review]

**Comment by jborgers** [comment]
> spring set 01 remarks:
> 
> `AvoidModelMapAsRenderParameter` - unit tests not complete (e.g. non-matches missing, second part of xpath expression is not covered) and what is comment for two bad ones: "[--> NOT, see JPCC-88?]"

see inline

> 
> `AvoidSpringApplicationContextRecreationTest` - assume creation of all ApplicationContexts is bad in methods - does not work in designer without proper classpath imports

ok, if works in unit tests, fine

> 
> PMD7 question: Why `MethodDeclaration`,`LocalVariableDeclaration`,`FieldDeclaration` but `VariableDeclarator`?

Maybe historically? Going from 6 to 7 and forgot to be consistent?

**Comment by stokpop** [comment]
> AvoidModelMapAsRenderParameter - unit tests not complete (e.g. non-matches missing, second part of xpath expression is not covered) and what is comment for two bad ones: "[--> NOT, see JPCC-88?]"

ok, will create issue to look at this later, let's first finish the others

>> PMD7 question: Why MethodDeclaration,LocalVariableDeclaration,FieldDeclaration but VariableDeclarator?

> Maybe historically? Going from 6 to 7 and forgot to be consistent?

We can ask in PMD7 after migration 🙂

**Comment by stokpop** [comment]
See: https://github.com/jborgers/PMD-jPinpoint-rules/issues/373

**Pull request merged**